storage_service/locations/models/s3.py

```diff
@@ -80,6 +80,19 @@ def browse(self, path):
         # strip leading slash on path
         path = path.lstrip('/')
 
+        # We need a trailing slash on non-empty prefixes because a path like:
+        #
+        #      /path/to/requirements
+        #
+        # will happily prefix match:
+        #
+        #      /path/to/requirements.txt
+        #
+        # which is not the intention!
+        #
+        if path != '':
+            path += '/'
```

Comment:
This is better, but I actually think it has to be `path = path.rstrip('/') + '/'`. The way it is, if the user tries to browse a path that ends in a forward slash, an extra slash will get added and nothing will be matched. The line just suggested fixes this.